### PR TITLE
plhash: Fix hash function for case-insensitive hashes

### PR DESCRIPTION
**Pull request by nirbhayc**
Signed-off-by: Nirbhay Choubey <nchoubey@bloomberg.net>

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/18e632e/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/18e632e/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: <b>1/380 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>ruleset_no_fp_generated [core dumped]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/18e632e/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/18e632e/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2564'>Resubmit</a></p>

File: util/plhash.c [hunk 1]
```diff
@@ -927,6 +927,24 @@ static unsigned int hash_default_strlen(const unsigned char *key,
...
 
+static unsigned int hash_default_strcaselen(const unsigned char *key,
+                                            int len __attribute_unused__)
+{
+    unsigned hash = 0;
+    uint8_t k = *key;
+
+    while (k) {
+        /* Convert to lowercase */
+        if ((k > 64) && (k < 91)) {
```
Comment (riverszhang89) [inline util/plhash.c:938]: This will not work for the extended ASCII codes (128 - 255), for instance, ã <-> Ã. Maybe just call `tolower()` here.

Comment (nirbhayc) [reply util/plhash.c]: Sure, done.

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/6d0c443/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/6d0c443/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/6d0c443/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2564'>Resubmit</a></p>

**Review by riverszhang89 — APPROVED** [review]
Good to merge.

**Comment by nirbhayc** [comment]
Thank you @riverszhang89!

**Pull request merged**